src/sources/fluent/mod.rs

```diff
@@ -445,16 +503,24 @@ impl From<FluentFrame> for SmallVec<[Event; 1]> {
     }
 }
 
-impl From<FluentEvent> for LogEvent {
+impl From<FluentEvent<'_>> for LogEvent {
     fn from(frame: FluentEvent) -> LogEvent {
         let FluentEvent {
             tag,
             timestamp,
             record,
+            log_namespace,
         } = frame;
 
         let mut log = LogEvent::default();
-        log.insert(log_schema().timestamp_key(), timestamp);
+
+        log_namespace.insert_vector_metadata(
+            &mut log,
+            log_schema().timestamp_key(),
+            "ingest_timestamp",
+            timestamp,
+        );
+
         log.insert("tag", tag);
         for (key, value) in record.into_iter() {
             log.insert(event_path!(&key), value);
```

Comment:
I guess these also need to be handled in the schema

Reply:
Ah good catch!

Reply:
though... the ones from `record`, are added dynamically and we don't know what the keys will be...

Reply:
`aws_s3` has similarly unknown/dynamic entries: https://github.com/vectordotdev/vector/blob/master/src/sources/aws_s3/mod.rs#L161-L171

Reply:
addressed in https://github.com/vectordotdev/vector/pull/15206/commits/eaef11960ef305327ed73ab6f31c0c6fb3e66934